internal/pkg/agent/application/upgrade/step_mark.go

```diff
@@ -277,11 +278,50 @@ func CleanMarker(log *logger.Logger, dataDirPath string) error {
 // LoadMarker loads the update marker. If the file does not exist it returns nil
 // and no error.
 func LoadMarker(dataDirPath string) (*UpdateMarker, error) {
-	return loadMarker(markerFilePath(dataDirPath))
+	return loadMarker(markerFilePath(dataDirPath), true)
 }
 
-func loadMarker(markerFile string) (*UpdateMarker, error) {
-	markerBytes, err := readMarkerFile(markerFile)
+func UpdateMarkerFile(markerFile string, opts ...func(*UpdateMarker)) error {
```

Comment:
More than options I think we are dealing with updater functions here
```suggestion
type markerUpdater func(*UpdateMarker)

func UpdateMarkerFile(markerFile string, updates ...markerUpdater) error {
```